jvm-packages/xgboost4j-gpu/src/main/java/ml/dmlc/xgboost4j/gpu/java/ColumnDMatrix.java

```diff
@@ -0,0 +1,96 @@
+/*
+ Copyright (c) 2021 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.gpu.java;
+
+import java.util.Iterator;
+
+import ml.dmlc.xgboost4j.java.DMatrix;
+import ml.dmlc.xgboost4j.java.XGBoostError;
+
+/**
+ * DMatrix based on GPU
+ */
+class ColumnDMatrix extends DMatrix {
+
+  /**
+   * Create DeviceQuantileDMatrix from iterator based on the cuda array interface
+   * @param iter     the cudf table batch to provide the corresponding cuda array interface
+   * @param missing  the missing value
+   * @param maxBin   the max bin
+   * @param nthread  the parallelism
+   * @throws XGBoostError
+   */
+  public ColumnDMatrix(Iterator<GpuTable> iter, float missing, int maxBin, int nthread)
+      throws XGBoostError {
+    super(0);
+    long[] out = new long[1];
+    Iterator<TableInfo> batchIter = new TableInfo.TableInfoBatchIterator(iter);
+    XGBoostJNI.checkCall(XGBoostJNI.XGDeviceQuantileDMatrixCreateFromCallback(
+        batchIter, missing, maxBin, nthread, out));
+    handle = out[0];
+  }
+
+  /**
+   * Create the normal DMatrix from column array interface
+   * @param featureArrayInterfaces the cuda array interface of feature columns
+   * @param missing missing value
+   * @param nthread threads number
+   * @throws XGBoostError
+   */
+  public ColumnDMatrix(String featureArrayInterfaces, float missing, int nthread)
```

Comment:
Do we need to expose this?

Reply:
Yeah, most of the real cases may not include base_margin or weight.

Reply:
I meant do we need to expose an interface that accepts string instead of real data?

Reply:
Yeah, better to expose this. then users can use this API to construct DMatrix from array interface

Reply:
I meant, why not

``` java
public ColumnMatrix(Table data, float missing) {
    String str = getArrayInterface(data)
    ...
}
```

Reply:
done